### DISCUSSION TITLE
Automerge isn't merging a PR

### DISCUSSION THREAD

**Discussion by darlenya** [renovatebot/renovate#11427] [Request Help]
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

26.7.0

### Describe the bug

A PR is not closed by automerge:

Command to reproduce the behaviour:
`npx renovate darlenya/renovate-demo-issue --token X9U6q4AMxmc-XK8-rZsV`

Here ist the config I used:

```
{
  "autodiscover": false,
  "baseBranches": ["main"],
  "endpoint": "https://gitlab.com/api/v4/",
  "labels": ["renovate", "dependencies", "automation"],
  "hostRules": [],
  "packageRules": [
    {
      "semanticCommitType": "fix",
      "labels": ["renovate", "dependencies", "automation", "depCi", "npm"],
      "prPriority": 0,
      "matchDepTypes": ["dependencies"]
    },
    {
      "semanticCommitType": "chore",
      "labels": ["renovate", "dependencies", "automation", "devChore", "npm"],
      "prPriority": -5,
      "matchDepTypes": ["devDependencies"]
    }
  ],
  "platform": "gitlab",
  "rangeStrategy": "pin",
  "recreateClosed": true,
  "rollbackPrs": true,
  "semanticCommitType": "fix",
  "semanticCommits": "enabled",
  "unicodeEmoji": true,
  "automerge": true,
  "gitLabAutomerge": true
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: getBranchStatus(renovate/main-eslint-7.x) (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
DEBUG: Got res with 0 results (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
DEBUG: getBranchStatus() result (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
       "branchStatus": "yellow",
       "branchName": "renovate/main-eslint-7.x"
DEBUG: Merge Request #3 does not need updating (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
DEBUG: PR is configured for automerge (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
DEBUG: PR is not ready for merge (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
       "canMergeReason": undefined
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction repository in the bug description

**Comment by HonkingGoose** [comment]
Please format your Renovate config. What you have right now is hard to read for us.

You can put your config in the online Prettier tool: https://prettier.io/playground/

Be sure to select `--parser` : `json` in the Prettier playground sidebar.

**Comment by darlenya** [comment]
In the edit mode it is formatted. When I do 'insert code' it is shown as before. Now I removed this and the spaces are removed.

**Comment by rarkins** [comment]
@darlenya you need to surround code blocks in trip backticks ```, so I have done that now

**Comment by darlenya** [comment]
Ok, thanks

**Comment by rarkins** [comment]
I think your title is wrong. Don't you mean "automerge isn't merging" instead of "closing"?

**Comment by rarkins** [comment]
Also "I have linked to a minimal reproduction repository in the bug description" seems incorrect because https://gitlab.com/darlenya/renovate-demo-issue either doesn't exist or is private

**Comment by rarkins** [comment]
Anyway your logs seem to tell you the problem anyway and this is not a bug.

```
DEBUG: getBranchStatus(renovate/main-eslint-7.x) (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
DEBUG: Got res with 0 results (repository=darlenya/renovate-demo-issue, branch=renovate/main-eslint-7.x)
```

There are no status checks reported from that branch. Please read https://docs.renovatebot.com/automerge-configuration/

In future please create a discussion first and not issues

**Reply by rarkins** [reply]
You should read all of that section, but the answer is in "Absence of tests"

**Reply by darlenya** [reply]
Hi, the repo is now public. Sorry for that. The automerge of our company repositories worked for over two years, but had stopped working since a few updates now. I have tried to create a repo to show the same behaviour as our company internal repos.

> You should read all of that section, but the answer is in "Absence of tests"

The repo has a test. Is it important that it must be test stage in the CI? This was not necessary before. In the demo repo the stage is named 'test' any how.

**Reply by rarkins** [reply]
The source for the GitLab getBranchStatus() is here: https://github.com/renovatebot/renovate/blob/d260dbf2d970b2d097b45b59cb3067ac532916a7/lib/platform/gitlab/index.ts#L386-L415 and https://github.com/renovatebot/renovate/blob/d260dbf2d970b2d097b45b59cb3067ac532916a7/lib/platform/gitlab/index.ts#L332-L353

I do not see any stage filtering.

Your config also shows that you are using the `gitlabAutomerge` feature. In this case don't you expect GitLab to perform the automerging and not Renovate?

**Reply by darlenya** [reply]
It seams that the issue is that the test is not running after the PR was created. So the status for this branch ist not there. I have now changed the the gitlab-ci.yml so that the test is running after each commit on each branch and the Pos where merged. So you are right it was not a problem from renovate.  Sorry for taken your time. And thanks for your help.
